### PR TITLE
Improve __uspapi behavior in iframe

### PR DESCRIPTION
**Pull request by ReinoutStevens**
## Type of change

- [x] Bugfix

## Description of change

This fixes the following two issues:

* https://github.com/prebid/Prebid.js/issues/6051
* https://github.com/prebid/Prebid.js/issues/6050

I've aligned the locator code with the `consentManagement.js` module so that use similar logic in finding where the `uspapi` function resides.

I've removed the code that removes the event listener after the consent information has been retrieved, as this breaks the `__uspapi` function that we installed on the current window. As a result, external libraries would think that the current window has a uspapi, call it, but never get a response.

## Other information

The existing unit tests already cover the cases of finding the uspapi in an iframe and directly, and still pass.
I've also tested this in a real world setup, and both the case of accessing the api directly and cross iframe work.

### CODE REVIEW

**Comment by ReinoutStevens** [comment]
@robertrmartinez would be a shame if this PR got closed due to inactivity

**Review by robertrmartinez — APPROVED** [review]
Looks fine to me, although I was not the one who implemented the iframe lookup initially.

Needs a second review anyways, so going to assign the second review to @jsnellbaker since he wrote the initial consentManagement code!

**Comment by smenzer** [comment]
@jsnellbaker can we get your 2nd review on this one, please?

**Review by jsnellbaker — APPROVED** [review]
LGTM

**Pull request merged**
